ttnn/cpp/ttnn/operations/moreh/moreh_dot_op_backward/moreh_dot_backward.hpp

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#include "ttnn/decorators.hpp"
+namespace ttnn::operations::moreh::moreh_dot_backward {
+struct MorehDotBackward {
+    static std::vector<Tensor> invoke(
+        const Tensor &output_grad,
+        const Tensor &input,
+        const Tensor &other,
+        std::optional<const Tensor> input_grad,
+        std::optional<const Tensor> other_grad,
+        const MemoryConfig &mem_config);
+};
+}
+
+namespace ttnn {
+constexpr auto moreh_dot_backward =
+    ttnn::register_operation<"ttnn::moreh_dot_backward", ttnn::operations::moreh::moreh_dot_backward::MorehDotBackward>();
```

Comment:
```suggestion
    ttnn::register_operation_with_auto_launch_op<"ttnn::moreh_dot_backward", ttnn::operations::moreh::moreh_dot_backward::MorehDotBackward>();
```

Reply:
I thought register_operation_with_auto_launch_op is not usable in case of the op return value is a vector of tensors?

Reply:
You can ignore the requests made in this comment. 
Since the review of the PR fixing the build failure issue is taking time, it has been decided to use `register_operation` for now.